### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by ngeelani48**
**In this pull request @ngeelani48 and @Koshao-sys added following requirements:**

- Installed CanCanCan in the project.
- Added a role column to the users table.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.
  - Added the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.
  - Added the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def show
```
Comment (GzimAsani) [inline app/controllers/posts_controller.rb:13]: - [ ] On top of your file use CanCanCan for Post authorization
 
```
  class PostsController < ApplicationController
    load_and_authorize_resource

```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @ngeelani48 and @Koshao-sys                          :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the ability.rb module+1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (ngeelani48) [reply app/controllers/posts_controller.rb:13]: Hi, @GzimAsani Thanks for the review 💯
I forgot to do it. Thanks for mentioning it 👍

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Applied the reviewer's suggestion properly

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**